src/ripple/protocol/STAmount.h

```diff
@@ -377,15 +377,29 @@ divide (STAmount const& v1, STAmount const& v2, Issue const& issue);
 STAmount
 multiply (STAmount const& v1, STAmount const& v2, Issue const& issue);
 
-// multiply, or divide rounding result in specified direction
+/** Control when bugfixes that require switchover dates are enabled */
+class STAmountCalcSwitchovers
+{
+    bool enableUnderflowFix_ {false};
+  public:
+    explicit
+    STAmountCalcSwitchovers (std::uint32_t parentCloseTime);
+    STAmountCalcSwitchovers () = default;
```

Comment:
The default constructor makes me nervous.  My spider sense says the default constructor is a little too easy to abuse.  In the future while someone is maintaining code it would be easy to either forget to put in the date or, if the date is difficult to get ahold of, they could "just use the default constructor" and get bad behavior.